dts/bindings/adc/st,stm32-adc.yaml

```diff
@@ -44,5 +44,26 @@ properties:
     type: boolean
     description: Indicates if the ADC has a dedicated internal vbat monitoring channel.
 
+  temp-channel:
```

Comment:
The new properties here seem to make the previous has-xxx properties redundant.  It's also possible to create an invalid dts that doesn't produce a warning.  E.g.

```
/* no has-vref-channel */
vref-channel = <10>;
```
Or:
```
has-vref-channel;
has-vbat-channel;
/* no vref-channel nor vbat-channel.  So they are both on channel 0? */
```

If the backward compatibility rules don't make it too onerous, you could get rid of the old has-xx properties.  In the code, use:
```
.vref_channel = DT_INST_PROP_OR(index, vref_channel, 0xff),
```
Where 0xff is a code to indicate no vref channel.

You could also keep boolean fields in the config struct, but they seem unnecessary, like this:
```
.has_vref_channel = DT_INST_NODE_HAS_PROP(index, vref_channel),
.vref_channel = DT_INST_PROP_OR(index, vref_channel, 0),
```

Reply:
You're right, it is cleaner this way. Done